### PR TITLE
Ch6 - account for alternative solutions for Action Multiply Int

### PR DESCRIPTION
**Pull request by gomain**
As discussed in #334 and raised in #198. This PR follows this [proposal](https://github.com/purescript-contrib/purescript-book/pull/334#issuecomment-841630193).

This would void #335.

### CODE REVIEW

File: exercises/chapter6/test/Main.purs [hunk 1]
```diff
@@ -180,7 +181,8 @@ Note to reader: Delete this line to expand comment block -}
           Assert.equal (act m1 (act m2 a))
             $ act (m1 <> m2) a
         test "concrete" do
-          Assert.equal 15
+          Assert.assert "not one of [ 15, 125 ]"
+            $ flip elem [ 15, 125 ]
             $ act m1 a
```
Comment (milesfrain) [inline exercises/chapter6/test/Main.purs:186]: Downside of this change is that we lose the helpful "expected x, got y" test failure messages.

Comment (gomain) [reply exercises/chapter6/test/Main.purs]: A reader wouldn't know what to expect. But fixed for consistency anyway.

**Comment by gomain** [comment]
In response to  https://github.com/purescript-contrib/purescript-book/pull/334#issuecomment-841735416, The additional exercise could be
> (Difficult) There are actually multiple ways to implement an instance of `Action Multiply Int`. How many can you think of? Purescript does not allow multiple implementations of a same instance, so you will have to replace your original implementation. _Hint_: the test already covers 3 implementations.

**Comment by milesfrain** [comment]
Looks good. Would you like to also add the bonus exercise to the text? Thinking we should clarify to:
> Hint: The test already covers 3 unique implementations.

(just added the word "unique").

And what about removing the "Does this instance satisfy the laws listed above?" lines from the exercise texts? Those lines were written before we had the TDD framework.

It might also be helpful to be a bit more specific in what we're asking for in the first exercise. Perhaps add this clarification to so beginners have something more concrete to start with (similar to the phrasing of the next `String` exercise).

> Write an instance which implements this action which adds an integer to itself some number of times:
```instance actionMultiplyInt :: Action Multiply Int```

I think it's also reasonable to accept the `act _ m = m` solution, since it is technically law abiding, but up to you whether you want to include it in the tests.

**Comment by gomain** [comment]
> Hint: The test already covers 3 unique implementations.

I don't think we need to stress the _unique_ nature of implementations. We already stated multiple implementations are not allowed.

> And what about removing the "Does this instance satisfy the laws listed above?"

Changed this line to an instructive "Verify that this instance satisfies the laws listed above.". This _is_ the criterion of this exercise.

> _Hint:_ This action should add an integer to itself some number of times.

Omitting such a hint allows readers to explore implementations that _don't_ satisfy the laws. Given the abstractness of typeclasses, a little math won't hurt.

> I think it's also reasonable to accept the `act _ m = m` solution

I agree with concerns raised in #198, a reader may _accidentally_ come up with this solution by just satisfying the types. There is a [comment section](https://github.com/purescript-contrib/purescript-book/pull/338/files#diff-d828757a85d8476587a839f570b0b5412ba065a2b5121b0a6cd40e6dc40dac9aR219-R224) that addresses this solution.

**Comment by gomain** [comment]
I also altered the text, adding some elaboration on the laws.

**Pull request merged**